tests/auth_tests/test_forms.py

```diff
@@ -377,6 +378,18 @@ def test_unicode_username(self):
         self.assertTrue(form.is_valid())
         self.assertEqual(form.non_field_errors(), [])
 
+    @override_settings(AUTH_USER_MODEL='auth_tests.CustomUserWithLongerUsernameField')
+    def test_username_field_max_length_matches_user_model(self):
+        # MySQL VARCHAR allows up to 255 characters, which is more than the
+        # max length allowed by AuthenticationForm with the default user model
```

Comment:
I think this comment is unnecessary, but we can leave it to the final review.

Reply:
Agreed, it seems a bit out of place. Also, I think we could use an existing user model:
```python
@override_settings(AUTH_USER_MODEL='auth_tests.CustomEmailField')
def test_username_field_max_length_matches_user_model(self):
    self.assertEqual(CustomEmailField._meta.get_field('username').max_length, 255)
    data = {
        'username': 'u' * 255,
        'password': 'pwd',
        'email': 'test@example.com',
    }
    CustomEmailField.objects.create_user(**data)
    form = AuthenticationForm(None, data)
    self.assertEqual(form.fields['username'].max_length, 255)
    self.assertEqual(form.errors, {})
```
I'd also like to see a test that fails if the `254` fallback value in the `getattr()` is changed.